### PR TITLE
discoveryserver: fixes deployment yaml and adds notes about use of Eureka's port

### PR DESCRIPTION
**Pull request by codefromthecrypt**
tried the docs, which drifted a bit.

Note: My brain broke about why this is using the same port as eureka. It seems it was an arbitrary choice, not a transition story. So, I didn't document anything about that.

### CODE REVIEW

File: docs/modules/ROOT/pages/spring-cloud-kubernetes-discoveryserver.adoc [hunk 1]
```diff
@@ -204,3 +204,7 @@ items:
 
 
 ----
+
+*Note* While the port is the same as Eureka's default (8761), there is no conflict despite the JSON
```
Comment (codefromthecrypt) [inline docs/modules/ROOT/pages/spring-cloud-kubernetes-discoveryserver.adoc:208]: ps wild guess this was chosen intentionally, like so custom servers can mount both endpoints and you only need to manage one discovery pod. If anyone knows the trivia why, do tell! cc @ryanjbaxter @spencergibb

Comment (ryanjbaxter) [reply docs/modules/ROOT/pages/spring-cloud-kubernetes-discoveryserver.adoc:208]: IDK why you would be using two different discovery servers

Comment (codefromthecrypt) [reply docs/modules/ROOT/pages/spring-cloud-kubernetes-discoveryserver.adoc:208]: on my side, it was a guess wondering why this is using the eureka port. I don't know the reason, so the important part was the question not my guess of the answer ;) If no one remembers, no worries.

Comment (codefromthecrypt) [reply docs/modules/ROOT/pages/spring-cloud-kubernetes-discoveryserver.adoc:208]: I know of at least one site trying to migrate from eureka to not eureka. That said, maybe the note is more trivia and calling attention to choosing the same well-known port as something else isn't helpful. I'll remove it!

Comment (codefromthecrypt) [reply docs/modules/ROOT/pages/spring-cloud-kubernetes-discoveryserver.adoc]: removed it!

**Pull request merged**